test/unit-test/CMakeLists.txt

```diff
@@ -205,3 +205,42 @@ create_test(${utest_name}
             "${utest_dep_list}"
             "${test_include_directories}"
         )
+
+# Create sara_r4 utest
+set( CELLULAR_MODULE_SARA_R4_SOURCE "${MODULE_ROOT_DIR}/modules/sara_r4" )
+set( cellular_sara_r4_real_name "${project_name}_module_sara_r4_real" )
+
+# list the files you would like to test here
+list(APPEND cellular_module_sara_r4_real_source_files
+            ${CELLULAR_MODULE_SARA_R4_SOURCE}/cellular_r4.c
+            ${CELLULAR_MODULE_SARA_R4_SOURCE}/cellular_r4_wrapper.c
+            ${CELLULAR_MODULE_SARA_R4_SOURCE}/cellular_r4_api.c
+            ${CELLULAR_MODULE_SARA_R4_SOURCE}/cellular_r4_urc_handler.c
+        )
+
+create_real_library(${cellular_sara_r4_real_name}
+                    "${cellular_module_sara_r4_real_source_files}"
+                    "${real_include_directories}"
+                    ""
+        )
+
+set(utest_link_list "")
+list(APPEND utest_link_list
+            lib${cellular_common_real_name}.a
+            lib${cellular_sara_r4_real_name}.a
+        )
+
+set(utest_dep_list "")
+list(APPEND utest_dep_list
+            ${cellular_common_real_name}
+            ${cellular_sara_r4_real_name}
+        )
+
+set(utest_name "${project_name}_sara_r4_utest")
+set(utest_source "${project_name}_sara_r4_utest.c")
```

Comment:
I saw there is an empty file `cellular_sara_r4_utest.c` to do warning check, I think your purpose is to build the module code, is it necessary to add the empty file for this purpose?

Reply:
Move the module compilation CI check to separate PR.

Reply:
Sure, that's what we discuss before.